include/unitTestMain.h

```diff
@@ -21,13 +21,51 @@
 #include <cppunit/TestResultCollector.h>
 #include <cppunit/TestRunner.h>
 
+#include <memory>
+#include <stdexcept>
+
+/// implements test program's main() function while enabling customization
+class TestProgram
+{
+public:
+    virtual ~TestProgram() = default;
+
+    /// Runs before all tests.
+    /// Does nothing by default.
+    virtual void startup() {}
+
+    /// Runs after all tests, regardless of their outcome.
+    /// Does nothing by default.
+    virtual void shutdown() {}
+
+    /// Implements main(), combining all the steps.
+    /// Must be called from main().
+    /// \returns desired main() result.
+    int run();
+
+private:
+    bool runTests();
+};
+
 int
-main(int, char *[])
+TestProgram::run()
 {
 #if ENABLE_DEBUG_SECTION
     Debug::Levels[ENABLE_DEBUG_SECTION] = 99;
 #endif
 
+    startup();
+    const auto result = runTests();
```

Comment:
What if runTests() throws an exception? Is there any risk that we will leak system resources (e.g. shm segments)?
Would it make sense to catch, shutdown, rethrow? Or to extend the API to have an explicit on-exception handler? Or could that role be handled by the virtual descriptor, and in that case we should document it?

Reply:
> Your question made me think that adding shutdown() method was premature. I will be removing that method.

Done in commit c69f505.

Reply:
It is out-of-scope for this PR, but perhapse we should have a team discussion on redesigning these to be proper unit tests instead of the smoke-tests they currently are.